src/test/java/org/swisspush/reststorage/mocks/FailFastVertxAsyncFile.java

```diff
@@ -103,13 +128,38 @@ public AsyncFile setReadPos(long l) {
         throw new UnsupportedOperationException(msg);
     }
 
+    @Override
+    public AsyncFile setReadLength(long readLength) {
+        throw new UnsupportedOperationException(msg);
+    }
+
+    @Override
+    public long getReadLength() {
+        return 0;
```

Comment:
A "*fail-fast*" class is responsible to fail for whatever call is made. Therefore this method should "*fail-fast*".

(*Occurs multiple times*)